File: horizontal-picker/src/main/java/com/github/jhonnyx2012/horizontalpicker/HorizontalPickerAdapter.java
package com.github.jhonnyx2012.horizontalpicker;


import android.app.AlarmManager;
import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.Drawable;
import android.support.v4.graphics.drawable.DrawableCompat;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.joda.time.DateTime;

import java.util.ArrayList;
import java.util.Date;

/**
 * Created by jhonn on 22/02/2017.
 */

public class HorizontalPickerAdapter extends RecyclerView.Adapter<HorizontalPickerAdapter.ViewHolder> {

    private static final long DAY_MILLIS = AlarmManager.INTERVAL_DAY;
    private int itemWidth;
    private final OnItemClickedListener listener;
    private ArrayList<Day> items;
    private boolean selectOnScroll;

    public HorizontalPickerAdapter(int itemWidth,
                                   OnItemClickedListener listener,
                                   Context context,
                                   int daysToCreate,
                                   int offset,
                                   boolean selectOnScroll) {
        items=new ArrayList<>();
        this.itemWidth=itemWidth;
        this.listener=listener;
        generateDays(daysToCreate, DateTime.now().minusDays(offset));
        this.selectOnScroll = selectOnScroll;
    }

    public void generateDays(int n, DateTime initialDate) {
        items.clear();

        for (int i = 0; i<n; i++) {
            DateTime date = new DateTime(initialDate).plusDays(i);
            items.add(new Day(date));
        }
    }


    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ViewHolder(
                LayoutInflater.from(parent.getContext())
                        .inflate(R.layout.item_day,parent,false));
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Day item = getItem(position);

        holder.tvDay.setText(item.getDay());
        holder.tvWeekDay.setText(item.getWeekDay());
        if(item.isSelected())
        {
            holder.tvDay.setBackgroundDrawable(getDaySelectedBackground(holder.itemView));
            holder.tvDay.setTextColor(Color.WHITE);
        }
        else if(item.isToday())
        {
            holder.tvDay.setBackgroundDrawable(getDayTodayBackground(holder.itemView));
            holder.tvDay.setTextColor(holder.tvDay.getResources().getColor(R.color.textColorPrimary));
        }
        else
        {
            holder.tvDay.setBackgroundColor(0);
            holder.tvDay.setTextColor(holder.tvDay.getResources().getColor(R.color.textColorPrimary));
        }
    }

    private Drawable getDaySelectedBackground(View view) {
        Drawable drawable=view.getResources().getDrawable(R.drawable.background_day_selected);
        DrawableCompat.setTint(drawable, view.getResources().getColor(R.color.colorPrimary));
        return drawable;
    }

    private Drawable getDayTodayBackground(View view) {
        Drawable drawable = view.getResources().getDrawable(R.drawable.background_day_today);
        return drawable;
    }

    public Day getItem(int position) {
        return items.get(position);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        TextView tvDay,tvWeekDay;

        public ViewHolder(View itemView) {
            super(itemView);
            tvDay= (TextView) itemView.findViewById(R.id.tvDay);
            tvDay.setWidth(itemWidth);
            tvWeekDay= (TextView) itemView.findViewById(R.id.tvWeekDay);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            listener.onClickView(v,getAdapterPosition());
        }
    }
}

File: horizontal-picker/src/main/kotlin/com/github/mcebular/horizontalpicker/HorizontalPickerAdapter.kt
package com.github.mcebular.horizontalpicker

import android.graphics.Color
import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.core.graphics.drawable.DrawableCompat
import androidx.recyclerview.widget.RecyclerView
import com.github.mcebular.horizontalpicker.databinding.ItemDayBinding
import java.time.LocalDate

class HorizontalPickerAdapter(
    private val listener: OnItemClickListener,
    private var itemWidth: Int,
    daysToCreate: Int,
    offset: Int
) : RecyclerView.Adapter<HorizontalPickerAdapter.ViewHolder>() {

    private val items: ArrayList<Day> = ArrayList()

    init {
        generateDays(daysToCreate, LocalDate.now().minusDays(offset.toLong()))
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = ItemDayBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    override fun getItemCount(): Int {
        return items.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = items[position]
        val binding = holder.binding

        binding.textViewDay.text = item.day
        binding.textViewWeekDay.text = item.weekDay

        if (item.selected) {
            binding.textViewDay.background = getDaySelectedBackground(binding.root)
            binding.textViewDay.setTextColor(Color.WHITE)
        } else if (item.isToday) {
            binding.textViewDay.background = getDayTodayBackground(binding.root)
            binding.textViewDay.setTextColor(ContextCompat.getColor(binding.root.context, R.color.textColorPrimary))
        } else {
            binding.textViewDay.setBackgroundColor(0)
            binding.textViewDay.setTextColor(ContextCompat.getColor(binding.root.context, R.color.textColorPrimary))
        }

    }

    fun getItem(position: Int): Day {
        return items[position]
    }

    private fun generateDays(n: Int, initialDate: LocalDate) {
        items.clear()
        for (i in 0 until n) {
            val date: LocalDate = initialDate.plusDays(i.toLong())
            items.add(Day(date))
        }
    }

    private fun getDaySelectedBackground(view: View): Drawable? {
        val drawable = ContextCompat.getDrawable(view.context, R.drawable.background_day_selected)
        val color = ContextCompat.getColor(view.context, R.color.colorPrimary)
        drawable?.let { DrawableCompat.setTint(it, color) }
        return drawable
    }

    private fun getDayTodayBackground(view: View): Drawable? {
        return ContextCompat.getDrawable(view.context, R.drawable.background_day_today)
    }

    inner class ViewHolder(internal var binding: ItemDayBinding) : RecyclerView.ViewHolder(binding.root), View.OnClickListener {

        init {
            binding.textViewDay.width = itemWidth
            itemView.setOnClickListener(this)
        }

        override fun onClick(v: View?) {
            listener.onClickView(v, bindingAdapterPosition)
        }
    }
}